pkg/controller.v1/common/job.go

```diff
@@ -155,6 +142,40 @@ func (jc *JobController) ReconcileJobs(
 		return nil
 	}
 
+	if trainutil.IsJobSuspended(runPolicy) {
+		if err = jc.CleanUpResources(runPolicy, runtimeObject, metaObject, jobStatus, pods); err != nil {
+			return err
+		}
+		for rType := range jobStatus.ReplicaStatuses {
+			jobStatus.ReplicaStatuses[rType].Active = 0
```

Comment:
I guess this would be set anyway by the other code once the replica pods are cleaned up. This is the approach we take in MPIJob and Job. I would like if we could apply here the same approach

Reply:
We don't have any other codes to reset the Active field, and the `replicaStatus[*].Active` is reset only by https://github.com/kubeflow/training-operator/blob/72f251290484b9da8570d8cbb7ed45bd2d228665/pkg/controller.v1/common/job.go#L143-L148.

So we need to reset the `Active` field here if the job is suspended.

However, since I think we should reset the `Active` field when cleaning up replica pods, I would do the refactoring in the follow-ups.

Reply:
is there any code that sets Active to non zero?

Reply:
> is there any code that sets Active to non zero?

Yes, here:  https://github.com/kubeflow/training-operator/blob/72f251290484b9da8570d8cbb7ed45bd2d228665/pkg/controller.v1/common/pod.go#L372 -> https://github.com/kubeflow/training-operator/blob/72f251290484b9da8570d8cbb7ed45bd2d228665/pkg/controller.v1/common/status.go#L15 -> https://github.com/kubeflow/training-operator/blob/72f251290484b9da8570d8cbb7ed45bd2d228665/pkg/core/status.go#L34-L50

Reply:
so those functions wouldn't be called in the next reconcile, essentially resetting the number of active pods?

Reply:
> so those functions wouldn't be called in the next reconcile

Yes. If the job is suspended, JobController never calls `ReconcilePods()`: https://github.com/tenzen-y/training-operator/blob/ce7259ecfaacbd529b6b1095dd6b632517dac0d0/pkg/controller.v1/common/job.go#L147-L173

Reply:
Instead of setting counts manually, why can't it be derived from the status of all pods? Since we have already cleaned up, active pods will be zero. We can do this refactoring separately as well.

Reply:
We don't have functions to decrease Active count, only have functions to reset the count. So I guess we should refactor ReconcileJob(). However, the refactor will affect Succeeded and Failed conditions, too. So I would like to work on another PR.